kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -372,7 +372,20 @@ impl<'tcx> GotocCtx<'tcx> {
                 // We want to return the function pointer (not to be confused with function item)
                 self.codegen_func_expr(instance, loc).address_of()
             }
-            GlobalAlloc::Static(def) => self.codegen_static_pointer(def),
+            GlobalAlloc::Static(def) => {
+                let int_def_id = rustc_internal::internal(self.tcx, def.def_id());
+                let anonymous = match self.tcx.def_kind(int_def_id) {
+                    rustc_hir::def::DefKind::Static { nested, .. } => nested,
+                    _ => false,
+                };
```

Comment:
Can we reasonably factor this out into a function like `is_anonymous_nested_static`? We now have that very same code in three places.